src/Core/ServerSettings.cpp

```diff
@@ -1128,6 +1129,32 @@ void ServerSettings::dumpToSystemServerSettingsColumns(ServerSettingColumnsParam
              {context->getRemoteReadThrottler() ? std::to_string(context->getRemoteReadThrottler()->getMaxSpeed()) : "0", ChangeableWithoutRestart::Yes}},
             {"max_remote_write_network_bandwidth_for_server",
              {context->getRemoteWriteThrottler() ? std::to_string(context->getRemoteWriteThrottler()->getMaxSpeed()) : "0", ChangeableWithoutRestart::Yes}},
+            {"max_io_thread_pool_size",
+             {getIOThreadPool().isInitialized() ? std::to_string(getIOThreadPool().get().getMaxThreads()) : "0", ChangeableWithoutRestart::Yes}},
```

Comment:
Maybe we should use setting.getValueString() here instead of 0 if it's not initialized. Although I'm not sure in which cases it isn't.

Reply:
Another option. Make it a std::optional<String> instead of a String, return std::nullopt and handle it in `res_columns[1]->insert`

Reply:
Or add these settings wich a check similar to `areBackgroundExecutorsInitialized`

Reply:
:+1:  If it's just clickhouse-local we can leave it as 0 then